File: src/main/java/com/example/demo/user/account/UserAccountConstraints.java
package com.example.demo.user.account;

public final class UserAccountConstraints {

    public static final int USERNAME_MAX_LENGTH = 15;
    public static final int NICKNAME_MAX_LENGTH = 30;

    public static final String USERNAME_TOO_LONG_MESSAGE =
            "Username must be at most " + USERNAME_MAX_LENGTH + " characters.";
    public static final String NICKNAME_TOO_LONG_MESSAGE =
            "Nickname must be at most " + NICKNAME_MAX_LENGTH + " characters.";

    private UserAccountConstraints() {
    }
}

File: src/main/java/com/example/demo/user/account/UserAccountConstraints.kt
package com.example.demo.user.account

object UserAccountConstraints {
    const val USERNAME_MAX_LENGTH: Int = 15
    const val NICKNAME_MAX_LENGTH: Int = 30

    const val USERNAME_TOO_LONG_MESSAGE: String = "Username must be at most $USERNAME_MAX_LENGTH characters."
    const val NICKNAME_TOO_LONG_MESSAGE: String = "Nickname must be at most $NICKNAME_MAX_LENGTH characters."
}
